compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -305,7 +306,7 @@ class ClassCodegen private constructor(
         if (field.origin != JvmLoweredDeclarationOrigin.CONTINUATION_CLASS_RESULT_FIELD) {
             val skipNullabilityAnnotations =
                 flags and (Opcodes.ACC_SYNTHETIC or Opcodes.ACC_ENUM) != 0 ||
-                        field.origin == JvmLoweredDeclarationOrigin.FIELD_FOR_STATIC_CALLABLE_REFERENCE_INSTANCE
+                        (field.origin == IrDeclarationOrigin.FIELD_FOR_OBJECT_INSTANCE && irClass.isSyntheticSingleton)
```

Comment:
Is there any test?

Reply:
There are plenty of bytecode listing tests with singleton lambdas, yes. I don't think it really matters whether we generate nullability annotations on their INSTANCE fields though, but whatever.

Reply:
Yes, indeed. Could you add test for optimization check?

Reply:
Sorry, I don't really understand what test you expect. Bytecode listing tests dump all annotations, so if there was a nullability annotation it'd be visible - is that not good enough?

Reply:
No, I've meant test for optimization itself that would check that it's singleton lambda (and yes most likely it would check also annotations).
Most likely it's enough to add // CHECK_BYTECODE_LISTING to already added new test

Reply:
Ah, got it. Done.